### DISCUSSION TITLE
OBS Studio connection problem

### DISCUSSION THREAD

**Discussion by snuth33** [obs-websocket-community-projects/obs-websocket-js#289] [Q&A]
Hi, I'm trying to connect to OBS Studio with `obs-websocket-js` but I can't. When I try, I get an OBS notification with a successful connection but after 2 seconds I get another notification for disconnection. I use :
- ` obs-websocket-js@5.0.0-beta.2`
- ` vite@1.0.0-rc.13`
- `vue@3.2.37`
- Chrome

When I open the console in Chrome, I have two messages:
- `WebSocket connection to 'ws://127.0.0.1:4444/' failed: createConnection @ obs-websocket-js.js:2828`
- `Uncaught (in promise) Error at OBSWebSocket.onClose (obs-websocket-js.js:3044:17)`

My code (file main.js):
`import OBSWebsocket from "obs-websocket-js";`
`const obs = new OBSWebsocket();`
`await obs.connect('ws://127.0.0.1:4444');`
`console.log('connected !!!');`

Can you help me ?
Thank you !

**Comment by t2t2** [comment]
My first guess would be check the port, in [obs-websocket 5.0.0-beta1](https://github.com/obsproject/obs-websocket/releases/tag/5.0.0-beta1) the default port was changed to 4455

**Reply by snuth33** [reply]
I change the port to 4455 and I have the same result. I get an OBS notification with a successful connection but after 2 seconds I get another notification for disconnection.

![image](https://user-images.githubusercontent.com/62243436/176964725-7cff6b10-7041-4f07-a7be-88aaead56280.png)

**Reply by t2t2** [reply]
Enable debug logging in obs-websocket and check the log file after connecting again (help -> log files -> view current log or upload current log file if to share it)

![image](https://user-images.githubusercontent.com/1122555/176965425-ec3beca0-3787-4361-b9b8-04cb7a950e99.png)

**Reply by snuth33** [reply]
I got this:
![image](https://user-images.githubusercontent.com/62243436/176966130-0596927e-884e-4378-b07b-480df76eebfe.png)

**Reply by t2t2** [reply]
Ok might be due to lack of password when connecting (there's a default randomly generated password since v5), add a catch around connect and see if it's in the error message:

```js
import OBSWebsocket from "obs-websocket-js";
const obs = new OBSWebsocket();
try {
  await obs.connect('ws://127.0.0.1:4455');
  console.log('connected !!!');
} catch (err) {
  console.error(err, err.message)
}
```

**Reply by snuth33** [reply]
The error message is:
![image](https://user-images.githubusercontent.com/62243436/176967299-a4b458b6-0a1b-403c-937c-8dfd28c2146c.png)

**Reply by t2t2** [reply]
Check the console with debug logs on:

```
localStorage.debug = '*'
```

(Then reload the page, and in dev tools make sure verbose logging is on)

![image](https://user-images.githubusercontent.com/1122555/177013051-f9cf9e63-1693-4bf1-8df0-75b11305f198.png)

**Reply by snuth33** [reply]
I need to put this line in my code ?

**Reply by t2t2** [reply]
can just paste into the console and then refresh the page
